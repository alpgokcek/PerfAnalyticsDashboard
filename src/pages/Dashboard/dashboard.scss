@import "../../styles/variables.scss";
@import "../../styles/placeholders.scss";

.dashboard {
  &__title {
    width: 100%;
    margin-bottom: 20px;
  }
  &__container {
    display: grid;
    grid-template-columns: 1fr;
    width: 100%;
  }
}
