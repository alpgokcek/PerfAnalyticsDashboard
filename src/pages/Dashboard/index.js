import React from "react";
import Layout from "../../components/Layout";
import Charts from "../../components/Charts";

import "./dashboard.scss";

const Dashboard = () => {
  return (
    <Layout data-test="dashboard-wrapper">
      <div className="dashboard__container" data-test="dashboard-container">
        <Charts data-test="charts-component" />
      </div>
    </Layout>
  );
};

export default Dashboard;
