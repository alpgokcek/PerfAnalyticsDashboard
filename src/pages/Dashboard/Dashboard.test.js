import React from "react";
import Enzyme, { shallow } from "enzyme";
import EnzymeAdapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

import { findByTestAttribute } from "../../test";

import Dashboard from "./index";
import { MOCK_METRIC } from "../../test/consts";

Enzyme.configure({ adapter: new EnzymeAdapter() });

const setup = (initialState = {}, props) => {
  return shallow(<Dashboard {...props} />);
};
jest.mock("react-chartjs-2", () => ({
  Line: () => null,
}));

describe("unit and snapshot tests for dashboard page", () => {
  let wrapper;
  beforeAll(() => {
    const mGetRandomValues = jest.fn().mockReturnValueOnce(new Uint32Array(10));
    Object.defineProperty(window, "crypto", {
      value: { getRandomValues: mGetRandomValues },
    });
    jest
      .useFakeTimers("modern")
      .setSystemTime(new Date("2021-05-23").getTime());
  });
  beforeEach(() => {
    wrapper = setup({ metricsReducer: { metrics: [MOCK_METRIC] } });
  });
  it("should render correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render non-empty component", () => {
    const inputComponent = findByTestAttribute(wrapper, "dashboard-wrapper");
    expect(inputComponent.length).toBe(1);
  });

  it("should render dashboard container", () => {
    const inputComponent = findByTestAttribute(wrapper, "dashboard-container");
    expect(inputComponent.children()).toHaveLength(1);
  });

  it("should render charts component", () => {
    const inputComponent = findByTestAttribute(wrapper, "charts-component");
    expect(inputComponent).toBeDefined();
  });
});
