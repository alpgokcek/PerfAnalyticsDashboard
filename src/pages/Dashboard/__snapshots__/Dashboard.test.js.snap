// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit and snapshot tests for dashboard page should render correctly 1`] = `
<Layout
  data-test="dashboard-wrapper"
>
  <div
    className="dashboard__container"
    data-test="dashboard-container"
  >
    <Connect(Charts)
      data-test="charts-component"
    />
  </div>
</Layout>
`;
